### PR TITLE
[chore] move internal tests to cmd/otelcontribcol

### PR DESCRIPTION
**Pull request by atoulme**
Part 2 of #19456, we now move tests to cmd/otelcontribcol.

### CODE REVIEW

File: cmd/otelcontribcol/exporters_test.go [hunk 1]
```diff
@@ -160,14 +159,6 @@ func TestDefaultExporters(t *testing.T) {
...
 			},
 		},
-		{
-			exporter: "parquet",
-			getConfigFn: func() component.Config {
-				cfg := expFactories["parquet"].CreateDefaultConfig().(*parquetexporter.Config)
-				cfg.Path = t.TempDir()
-				return cfg
-			},
-		},
```
Comment (atoulme) [inline cmd/otelcontribcol/exporters_test.go:170]: we don't ship parquet, so removing as imports don't work.

File: cmd/otelcontribcol/receivers_test.go [hunk 2]
```diff
@@ -96,16 +94,6 @@ func TestDefaultReceivers(t *testing.T) {
...
-		{
-			receiver: "azureblob",
-			getConfigFn: func() component.Config {
-				cfg := rcvrFactories["azureblob"].CreateDefaultConfig().(*azureblobreceiver.Config)
-				cfg.ConnectionString = "DefaultEndpointsProtocol=http;AccountName=accountName;AccountKey=accountKey==;BlobEndpoint=test"
-				cfg.EventHub.EndPoint = "DefaultEndpointsProtocol=http;SharedAccessKeyName=secret;SharedAccessKey=secret;Endpoint=test.test"
-				return cfg
-			},
-			skipLifecyle: true, // Requires Azure event hub to run
-		},
```
Comment (atoulme) [inline cmd/otelcontribcol/receivers_test.go:108]: we don't ship azureblobreceiver - removing that test.

File: cmd/otelcontribcol/receivers_test.go [hunk 3]
```diff
@@ -131,14 +119,6 @@ func TestDefaultReceivers(t *testing.T) {
...
 			skipLifecyle: true, // Requires UAA (auth) endpoint to run
 		},
-		{
-			receiver: "chrony",
-			getConfigFn: func() component.Config {
-				cfg := rcvrFactories["chrony"].CreateDefaultConfig().(*chronyreceiver.Config)
-				cfg.Endpoint = "udp://localhost:323"
-				return cfg
-			},
-		},
```
Comment (atoulme) [inline cmd/otelcontribcol/receivers_test.go:141]: we don't ship chronyreceiver - removing that test.

File: internal/components/components_test.go
Comment (atoulme) [inline internal/components/components_test.go:1]: this test was already moved to cmd/otelcontribcol, and therefore was deleted.

**Comment by TylerHelmuth** [comment]
Should we include all components committed in Contrib in these tests or only the ones we ship in the released image?  If someone were to run `make otelcrontribcol` its possible to build an image that includes components that aren't included by the build manifests.

**Comment by atoulme** [comment]
> Should we include all components committed in Contrib in these tests or only the ones we ship in the released image? If someone were to run `make otelcrontribcol` its possible to build an image that includes components that aren't included by the build manifests.

I think eventually we should try to cover any component of a given stability into our offering, but right now we don't test what we ship, so let's start there.

After that, I'd say to achieve beta status, maybe we can have a test utility that checks that the component can conform to lifecycle.

**Comment by dmitryax** [comment]
This one looks good to me. As the next step, should we make `configschema/cfgmetadatagen` dependent on `cmd/otelcontribcol` instead of copying the [component.go to configschema/cfgmetadatagen](https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/19456/files#diff-17ae270cca8c6a95ec899a89e1337afc4d2b542d167c3201fcc260585693f7c2) without tests?

If we are doing this just to remove the `/` module, we can also make `internal/components` a separate module as an option.

**Comment by atoulme** [comment]
That's next indeed, and notice I move slowly because the initial PR just copied internal/components/components.go into configschema - I think it might make more sense to have them depend on otelcontribcol, but the components factories are not exported right now.

**Comment by TylerHelmuth** [comment]
> it looks like we have several components that are not listed - should we enforce this going forward?

That would be my vote.  If we include it in the repository it seems reasonable to include it in our local dev/test build.

**Comment by atoulme** [comment]
@TylerHelmuth this PR no longer removes components - it adds a few to place under test.

**Review by dmitryax — APPROVED** [review]

**Pull request merged**